Test improvements

ltseeley opened the pull request.

vaporos commented:
It isn't clear that the changes in the tests accomplish the same thing as the original tests. The Original tests were organized around real-world examples of how one might use the API (or, rather, that was the intent). Its not obvious this has been translated into the replacement tests. It might help to clarify "X is covered by Y" and then enhance the fn doc to explictly state what is being covered by the test.

ltseeley commented:
> It isn't clear that the changes in the tests accomplish the same thing as the original tests. The Original tests were organized around real-world examples of how one might use the API (or, rather, that was the intent). Its not obvious this has been translated into the replacement tests. It might help to clarify "X is covered by Y" and then enhance the fn doc to explictly state what is being covered by the test.

I updated the commit message to indicate which new tests replace which old ones. I also added a multi-key, single-context test which I realized was missing in the new tests. Let me know if anything seems to need further clarification.

peterschwarz commented on libcylinder/src/key.rs:
```diff
@@ -93,6 +101,14 @@ impl PrivateKey {
+impl std::fmt::Debug for PrivateKey {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        f.debug_struct("PrivateKey")
+            .field("bytes", &self.as_hex())
```
Is logging the complete key a security concern? Especially if this is just for testing purposes

ltseeley replied:
I guess the debug/display implementations aren't needed for the tests; I can remove them for private key if this is a security issue.

ltseeley replied:
Removed the display/debug impls for private key and updated the commit
